doc/specs/fftpack.md

```diff
@@ -198,46 +191,42 @@ program demo_zfftb
 end program demo_zfftb
 ```
 
-## `fft`
+### `fft`
 
-### Description
+#### Description
 
 Computes the forward complex discrete fourier transform (the fourier analysis).   
 
-### Status
+#### Status
 
 Experimental.
 
-### Class
+#### Class
 
 Pure function.
 
-### Snytax
+#### Snytax
 
 `result = [[fftpack(module):fft(interface)]](x [, n])`
 
-### Argument
+#### Argument
 
 `x`: Shall be a `complex` and rank-1 array.
 This argument is `intent(in)`. 
 
 `n`: Shall be an `integer` scalar.
 This argument is `intent(in)` and `optional`.  
-The needed length of the `complex` sequence `c`.
+Defines the length of the Fourier transform. If `n` is not specified (the default) then `n = size(x)`. If `n <= size(x)`, `x` is truncated, if `n > size(x)`, `x` is zero-padded.
```

Comment:
Do you think somebody will use this optional `n`? It seems one can simply truncate or zero pad by hand before calling this function. That would simplify the code and interface.

Reply:
Yes, Fortran arrays are very easy-to-use, and it is easy to construct arrays (array slicing and extending arrays), such as:
```fortran
complex :: c(10)
print *, fft(x(2:3))
print *, fft([x, ((0.0,0.0), i=1, 10)])
```
I think to adapt to Fortran, the interface should be simplified. As far as I know, `scipy.fft`, `numpy.fft`, `matlab.fft` all have optional `n`, R language `fft` does not.

Reply:
Good point, I didn't realize that `numpy` had the `n` parameter: https://numpy.org/doc/stable/reference/generated/numpy.fft.fft.html, I wonder how often it is used.